src/providers/ModalProvider/index.tsx

```diff
@@ -19,33 +26,56 @@ export const ModalContext = createContext<ModalContext>(null);
 
 const ModalProvider = ({ children }: PropsWithChildren) => {
   const [currentModal, setCurrentModal] = useState<ReactNode | null>(null);
+  const [animation, setAnimation] = useState<Animation>('appear');
   const isOpen = !!currentModal;
 
-  const openModal = (modal: ReactNode) => setCurrentModal(modal);
+  const openModal = (modal: ReactNode) => {
+    setAnimation('appear');
+    setCurrentModal(modal);
+  };
 
-  const closeModal = () => setCurrentModal(null);
+  const closeModal = () => {
+    setAnimation('disAppear');
+    setTimeout(() => {
+      setCurrentModal(null);
+    }, 200);
+  };
```

Comment:
이렇게 핸들러 안에 `timeout`을 설정하면 리렌더링될 때마다 타이머가 초기화되지 않나요? 그래서 `ref`에 담아야 하는 걸로 알고 있는데요..
사실 단순 css를 위해서 js로 타이머를 거는 게 좋지는 않은 것 같아서..
```suggestion
  const closeModal = () => {
    setAnimation('disAppear');
  };
```
이렇게만 하고 애니메이션 걸려있는 컴포넌트에 `onAnimationEnd` 핸들러 달아서 `setCurrentModal(null);` 실행하면 좋을 것 같아요.

Reply:
`onAnimationEnd` 핸들러가 있다니,, 충격적이면서 좋네요 👍👍👍
수정해보도록 할게요!